### PR TITLE
Add unit tests for pkg/storage/chunk/prometheus_chunk.go

### PR DESCRIPTION
**Pull request by leizor**
Fixes https://github.com/grafana/mimir/issues/3767

#### Checklist

- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/storage/chunk/chunk_test.go [hunk 1]
```diff
@@ -101,18 +111,29 @@ func testChunkEncoding(t *testing.T, encoding Encoding, samples int) {
...
-	require.False(t, iter.Scan() == chunkenc.ValFloat)
+	require.True(t, iter.Scan() == chunkenc.ValNone)
 	require.NoError(t, iter.Err())
 
 	// Check seek works after unmarshal
 	iter = chunk.NewIterator(iter)
 	for i := 0; i < samples; i += samples / 10 {
-		require.True(t, iter.FindAtOrAfter(model.Time(i*step)) == chunkenc.ValFloat)
+		val := iter.FindAtOrAfter(model.Time(i * step))
+		require.True(t, val != chunkenc.ValNone)
```
Comment (krajorama) [inline pkg/storage/chunk/chunk_test.go:136]: [nit]

```suggestion
		require.NotEqual(t, chunkenc.ValNone, val)
```

File: pkg/storage/chunk/chunk_test.go [hunk 2]
```diff
@@ -186,14 +252,24 @@ func testChunkBatch(t *testing.T, encoding Encoding, samples int) {
...
-			require.EqualValues(t, int64((i+j)*step), batch.Timestamps[j])
-			require.EqualValues(t, float64(i+j), batch.SampleValues[j])
+		require.NotEqual(t, chunkenc.ValNone, iter.Scan())
+		var batch Batch
+		switch encoding {
+		case PrometheusXorChunk:
+			batch = iter.Batch(BatchSize, chunkenc.ValFloat)
+			for j := 0; j < batch.Length; j++ {
+				require.EqualValues(t, int64((i+j)*step), batch.Timestamps[j])
+				require.EqualValues(t, float64(i+j), batch.SampleValues[j])
```
Comment (krajorama) [inline pkg/storage/chunk/chunk_test.go:262]: [rant] not the fault of this PR, but Batch is not encapsulated so batch.Timestamps/batch.SampleValues will break when we fix Batch bug+performance

**Review by krajorama — APPROVED** [review]
LGTM, as agreed lets rebase and merge after mimir-prometheus is updated on the branch

**Comment by leizor** [comment]
Hmm...weird, I wonder why this closed itself?

**Comment by leizor** [comment]
Oh interesting, all these changes already made it into the `sparsehistogram` branch yesterday; I guess I'll consider this issue ✅  then! I'll follow-up and get those suggested changes in another PR.

**Pull request closed without merging**